src/platform/plugins/shared/esql/server/extensions_registry/index.test.ts

```diff
@@ -227,6 +227,43 @@ describe('ESQLExtensionsRegistry', () => {
     });
   });
 
+  // --- unsetRecommendedQueries tests ---
+
+  describe('unsetRecommendedQueries', () => {
+    const METRICS_INDEX_PATTERN = 'metrics-*';
+
+    beforeEach(() => {
+      availableDatasources = {
+        indices: [
+          { name: 'logs-2023' },
+          { name: 'logs-2024' },
+          { name: METRICS_INDEX_PATTERN },
+          { name: 'other_index' },
+        ],
+        data_streams: [],
+        aliases: [],
+      };
+    });
+
+    it('should remove recommended queries from the registry', () => {
```

Comment:
Can we add more tests here testing more complicated examples, different solutions etc?